string in_data
---
string out_data